All articles feature

hollabaq86 opened the pull request:
@emgreen33 @TylerSmall19 @NoahRGuy 👀  🍷

hollabaq86 committed:
Create articles#index page

https://github.com/TylerSmall19/cotton-tales

As a visitor to the Rabbit Hole, I can view a list of all articles made.

Added method within Article model to "stub" the body into a smaller summary for easy display
Added html to view a list of all articles

hollabaq86 committed:
merge master after pull request to resolve any potential conflicts

Merge branch 'master' into all-articles-feature

hollabaq86 committed:
Tweak to article stub method, for stylizing in rendering show view.

hollabaq86 committed:
Tweak to articles#show, added link to each article

hollabaq86 committed:
updated feature test file for articles#index page

I added a pending test so that eventually we can test that clicking to an article title leads you to that article's #show page.

hollabaq86 committed:
merging master to resolve any potential conflicts

Merge branch 'master' into all-articles-feature

simanthastew commented on app/views/articles/index.html.erb:
```diff
@@ -1 +1,11 @@
+
+<ul class='articles-list'>
+	<%	@articles.each do |article| %>
+		<li id="#{article.id}">
```
L6 should use print squigglies.

simanthastew commented on spec/features/visiting_articles_index_spec.rb:
```diff
@@ -1,10 +1,11 @@
+feature "visiting the article index page" do
+  scenario "The user can visit the page that lists all the articles" do
+    visit "/articles"
+    expect(page).to have_content "All Articles"
```
This only tests that a header is displayed, so the test name may be a bit misleading.

simanthastew reviewed (commented):
Not many changes. Was this all the code you had in the push?

hollabaq86 committed:
Tweaks to Article model and model tests

https://github.com/TylerSmall19/cotton-tales/pull/26

As a User and a Guest, I can see a list of all articles.

Edited name of method to shorten body of an article into a summary
Edited Index view file with new method name
Add test for method in article model spec file

hollabaq86 committed:
tweak to index view

hollabaq86 committed:
adjust tests

hollabaq86 committed:
fix article model spec tests

TylerSmall19 committed:
Merge branch 'master' into all-articles-feature
